### ISSUE TITLE
Duplicated for `mutable` and `Dual` for forward.

### ISSUE THREAD

**Issue by vchuravy** [wsmoses/Enzyme.jl#189]
Right now the user can accidentally pass in `Duplicated(1.0, 0.0)` to `autodiff`, which is odd and surprising since duplicated is for the shadow storage loacation and `0.0` is immutable and can't be changed on the Julia semantics level. We should use a separate type e.g. `Dual` for forward diff.

**Comment by ayeankit** [comment]
hey @vchuravy , Is this issue still pending? Can I work on it?

**Comment by vchuravy** [comment]
Yes, it should be straightforward but will require updating the documentation in many places.
